pydexec/tests/test_pysu.py

```diff
@@ -0,0 +1,138 @@
+    # gosu-t 1000:root '1000:0:0' '1000:root:root'
+    gosu_t('1000:root', '1000:0:0', '1000:root:root')
+    # gosu-t 1000:daemon "1000:$(id -g daemon):$(id -g daemon)" '1000:daemon:daemon'  # noqa: E501
+    gosu_t('1000:daemon', '1000:%s:%s' % (_id('-g', 'daemon'), _id('-g', 'daemon')), '1000:daemon:daemon')  # noqa: E501
+    # gosu-t games "$(id -u games):$(id -g games):$(id -G games)" 'games:games:games'  # noqa: E501
+    gosu_t('games', '%s:%s:%s' % (_id('-u', 'games'), _id('-g', 'games'), _id('-G', 'games')), 'games:games:games')  # noqa: E501
+    # gosu-t games:daemon "$(id -u games):$(id -g daemon):$(id -g daemon)" 'games:daemon:daemon'  # noqa: E501
+    gosu_t('games:daemon', '%s:%s:%s' % (_id('-u', 'games'), _id('-g', 'daemon'), _id('-g', 'daemon')), 'games:daemon:daemon')  # noqa: E501
+
+    # gosu-t 0: "0:0:$(id -G root)" "root:root:$(id -Gn root)"
+    gosu_t('0', '0:0:%s' % (_id('-G', 'root'),), 'root:root:%s' % (_id('-Gn', 'root'),))  # noqa: E501
+    # gosu-t '' "$(id -u):$(id -g):$(id -G)" "$(id -un):$(id -gn):$(id -Gn)"
+    gosu_t('', '%s:%s:%s' % (_id('-u'), _id('-g'), _id('-G')), 'root:root:%s' % (_id('-Gn', 'root'),))  # noqa: E501
+    # gosu-t ':0' "$(id -u):0:0" "$(id -un):root:root"
+    gosu_t(':0', '%s:0:0' % (_id('-u'),), '%s:root:root' % (_id('-un'),))
+
+    # [ "$(gosu 0 env | grep '^HOME=')" = 'HOME=/root' ]
+    assert_that(pysu_env_home('0'), Equals(['HOME=/root']))
+    # [ "$(gosu 0:0 env | grep '^HOME=')" = 'HOME=/root' ]
+    assert_that(pysu_env_home('0:0'), Equals(['HOME=/root']))
+    # [ "$(gosu root env | grep '^HOME=')" = 'HOME=/root' ]
+    assert_that(pysu_env_home('root'), Equals(['HOME=/root']))
+    # [ "$(gosu 0:root env | grep '^HOME=')" = 'HOME=/root' ]
+    assert_that(pysu_env_home('0:root'), Equals(['HOME=/root']))
+    # [ "$(gosu root:0 env | grep '^HOME=')" = 'HOME=/root' ]
+    assert_that(pysu_env_home('root:0'), Equals(['HOME=/root']))
+    # [ "$(gosu root:root env | grep '^HOME=')" = 'HOME=/root' ]
+    assert_that(pysu_env_home('root:root'), Equals(['HOME=/root']))
+    # [ "$(gosu 0:1000 env | grep '^HOME=')" = 'HOME=/root' ]
+    assert_that(pysu_env_home('0:1000'), Equals(['HOME=/root']))
+    # [ "$(gosu root:1000 env | grep '^HOME=')" = 'HOME=/root' ]
+    assert_that(pysu_env_home('root:1000'), Equals(['HOME=/root']))
+    # [ "$(gosu 1000 env | grep '^HOME=')" = 'HOME=/' ]
+    assert_that(pysu_env_home('1000'), Equals(['HOME=/']))
+    # [ "$(gosu 1000:0 env | grep '^HOME=')" = 'HOME=/' ]
+    assert_that(pysu_env_home('1000:0'), Equals(['HOME=/']))
+    # [ "$(gosu 1000:root env | grep '^HOME=')" = 'HOME=/' ]
+    assert_that(pysu_env_home('1000:root'), Equals(['HOME=/']))
+    # [ "$(gosu games env | grep '^HOME=')" = 'HOME=/usr/games' ]
+    assert_that(pysu_env_home('games'), Equals(['HOME=/usr/games']))
+    # [ "$(gosu games:daemon env | grep '^HOME=')" = 'HOME=/usr/games' ]
+    assert_that(pysu_env_home('games:daemon'), Equals(['HOME=/usr/games']))
```

Comment:
Can we break this up into smaller tests? We don't necessarily need one per assertion, but some kind of logical grouping would make it easier to reason about.